src/dynarec/rv64/dynarec_rv64_emit_tests.c

```diff
@@ -23,6 +23,69 @@
 #include "dynarec_rv64_functions.h"
 #include "dynarec_rv64_helper.h"
 
+// emit CMP8 instruction, from cmp s1, s2, using s3, s4, s5 and s6 as scratch
+void emit_cmp8(dynarec_rv64_t* dyn, int ninst, int s1, int s2, int s3, int s4, int s5, int s6) {
+    CLEAR_FLAGS();
+    IFX_PENDOR0 {
+        SB(s1, xEmu, offsetof(x64emu_t, op1));
+        SB(s2, xEmu, offsetof(x64emu_t, op2));
+        SET_DF(s4, d_cmp8);
+    } else {
+        SET_DFNONE(s4);
+    }
+
+    IFX(X_AF | X_CF | X_OF) {
+        // for later flag calculation
+        NOT(s5, s1);
+    }
+
+    // It's a cmp, we can't store the result back to s1.
+    SUB(s6, s1, s2);
```

Comment:
Don't we need an `ANDI(s6, s6, 0xff);` also?

Reply:
X_ZF also needs it.
And also might need to put back the value in a register,

It safer to be sure the value is 8bits only, so I think a systematic ANDI is better.

Reply:
> X_ZF also needs it.

Indeed, force-pushed.